### ISSUE TITLE
Autofocus with Arducam

### ISSUE THREAD

**Issue by myscho15** [ccrisan/motioneyeos#2820]
### Preliminary Docs

I confirm that I have read the [CONTRIBUTING](https://github.com/ccrisan/motioneyeos/blob/master/.github/CONTRIBUTING.md) guide before opening this issue.

I confirm that I have read the [FAQ](https://github.com/ccrisan/motioneyeos/wiki/FAQ) before opening this issue.

### motionEyeOS Version

I am running motionEyeOS version: Latest 20200606

### Board Model

I am using the following board/model:Raspberry PI 3B+

### Camera

I am using the following type of camera: Now is Raspberry PI HQ cam with 16mm and 12 - 50mm lens (works fine)

I want buy: Arducam 21MP IMX230

### Network Connection

My motionEyeOS unit is connected to the network via: WIFI

 Hay: i use Motioneyeos with hardware in description. 

I have question: i need use smaller camera module for my project, i found on internet this Arducam IMX230 with autofocus (autofocus is very important for me), can anyone help me if will this camera work - autofocus? Or can anyone help me find camera with autofocus for motioneye os? I dont need 21MP camera (in motioneyeos i setup 1920x1080 or bit more resolution), but i looking for camera with autofocus and good quality still images... Thank you very much for you answer and help.

**Comment by starbasessd** [comment]
Autofocus is a manufacturer's prerogative, with auto being the key word here. The best motioneye would be able to help with would be to over-ride the focus, and it wouldn't be automatic.
If there are commands that the camera would be able to accept, you could send them using the "Action Buttons" \( https://github.com/ccrisan/motioneye/wiki/Action-Buttons \)

**Comment by myscho15** [comment]
Hay thank you for your answer.

Sorry my English is poor and my IT skills are poor too.
I have my first setup and i still learning using the motioneyeos, i dont have friends with this IT skills, i know you are developers and IT experts like you. But i dont know where can i find help and firt i tried here.

First: with the Rpi HQ CAM i focusing manualy with hand - is not good - because i have camera with lens in box on tree. - i looking for something that can i focusing from motioneye os (for me was that autofocus - over ride focus)

Yes focus from software (motioneyeos) with actions buttons sounds good, but first can you please help me if the ARDUCAM IMX230 (or another) will work (i mean will i need drivers for this lens to make focus from motioneyos with action buttons) and if i need driver i dont know ho i install it.
How can i add the action buttons to camera frame and will the focusing from motioneye os will work. 

Here is the product sheet from the camera  - is posible to control focus with action buttons?

https://www.uctronics.com/download/Amazon/B0324_Manual.pdf


Thank you for you patient and help. I will very grateful to you.

**Comment by starbasessd** [comment]
Most security cameras (including your Arducam) are what is known as 'fixed focus'. There isn't any adjustment for focus available. The PiCam you purchased doesn't have a motorized lens, so would not be 'adjustable'. They do make adjustable, motorized cameras (those that are 'PTZ' might be a choice). 
To use them, though, they are more 'stand alone' and you connect motionEye to them using a stream they provide (http, or rtsp). You can see a LOT of possibilities in the iSpy database.
\( https://www.ispyconnect.com/cameras \)
I personally use d-link and foscam cameras, along with some rpis + PiCams, but I don't like to make recommendations. 
The link to the Action Buttons explains that you have Camera1 set up, then you create the script that does 'feature x', then you make a copy of the script calling it /etc/motioneye/action_1 with action_1 being the name of the action from the link, like up_1, or left_1 or preset1_1...
If motionEye sees such a script, it will automatically place the button on the screen for that camera. If you have more than 1 camera, you'd have to add copies of the script with the same name, but with the camera number 2, or 3, or whatever, at the end instead of the number 1.

**Comment by myscho15** [comment]
Hay, thank you for answer, its hard for me to understand, its confusing for me, but this is my fault... i must learn more. Thank you.

**Comment by starbasessd** [comment]
If you ask questions, I do my best to answer. 
If you need something clarified, ask. I'm here to help. 
I can't do what I don't have hardware to reproduce, though. All my cameras are 'fixed focus' (non-adjustable). They are in focus from 2 feet to infinity without adjustment.
As to programming, if you find a program that does what you want, I can point you in the right direction.

**Comment by myscho15** [comment]
Yeah now i understand, i have only RPI HQ cam now and this with lens i must make exact focus for the distance for good pictures, but i dont know that about small lenses and 2feet to infinity...

But i make project with stepper motors + 3D print and some controller over wifi to make DIY focusing kit for PI HQ cam lenses. I mean wil be better for me. 

But i am verre happy for you patiente. :-)

**Comment by starbasessd** [comment]
Some interesting info for you:
https://www.arducam.com/raspberry-pi-camera/autofocus/
https://forum.allaboutcircuits.com/threads/raspberry-pi-hq-camera-autofocus.172915/
